docs/troubleshooting.md

```diff
@@ -12,6 +12,21 @@ title: Troubleshooting
     - This configuration is only necessary for XCUITest driver v4.3.0 or lower.
 * `shake` is implemented via AppleScript and works only on Simulator due to lack of support from Apple
 
+
+## Clear the application local data explicitly for real devices
+
+iOS real device could have a situation that has application data locally but the application package is not on the device. It can occur an [offload application](https://discussions.apple.com/thread/254887240) state, cached application state or when an application installation fails. An example of the installation failure is `ApplicationVerificationFailed` by invalid provisioning profile.
```

Comment:
There might be a situation where application data is present on the device although application itself is not installed. This could happen if:
- The app is in offload state
- The application state is cached
- There was an unexpected failure while installing the app. An example of such failure is the `ApplicationVerificationFailed` which happens while installing an app signed with an invalid provisioning profile.

Under the circumstances above the application identifier won't be listed in the `mobile: listApps` output neither deleted by `mobile: isAppInstalled` API. Setting `fullReset` or `enforceAppInstall` capabilities to `true` won't help to clear this data too.

The only way to completely get rid of the cached app data is to call the `mobile: removeApp` API with the appropriate bundle identifier.

The driver automatically tries to resolve application installs that failed because of the obsolete data presence, although, if you explicitly ask it to not perform the app uninstall then consider calling `mobile: removeApp` beforehand.